frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form/ocs-storage-class-form.tsx

```diff
@@ -161,3 +189,264 @@ export const PoolResourceComponent: React.FC<ProvisionerProps> = ({ onParamChang
+  );
+
+  return (
+    <Form>
+      <FormGroup
+        fieldId="storage-class-encryption"
+        helperTextInvalid="This is a required field"
+        isRequired
+      >
+        <Checkbox
+          id="storage-class-encryption"
+          isChecked={checked}
+          label={<StorageClassEncryptionLabel />}
+          aria-label="Storage class encryption"
+          description="An encryption key for each Persistent volume (block only) will be generated."
+          onChange={isChecked}
+          className="ocs-storageClass-encryption__form-checkbox"
+        />
+
+        {checked && (
+          <>
+            <Card isFlat className="ocs-storageClass-encryption__card">
+              {!csiConfigMapLoaded && <EncryptionCardLoading />}
+              {csiConfigMapLoaded && csiConfigMap && !editKMS && !csiConfigMapLoadError ? (
+                <KMSDetails />
+              ) : (
+                <>
+                  <KMSConfigure
+                    state={state}
+                    dispatch={dispatch}
+                    className="ocs-storageClass-encryption"
+                  />
+                  <div className="ocs-install-kms__save-button">
+                    <ButtonBar errorMessage={errorMessage} inProgress={progress}>
+                      <ActionGroup>
+                        <Button variant="secondary" onClick={updateKMS} isDisabled={!validSave}>
+                          Save
+                        </Button>
+                        <Button variant="plain" onClick={cancelKMSUpdate}>
+                          Cancel
+                        </Button>
+                      </ActionGroup>
+                    </ButtonBar>
+                  </div>
+                </>
+              )}
+            </Card>
+            <Alert
+              className="co-alert"
+              variant="warning"
+              title="Encrypted PVs cannot be cloned, expanded or create snapshots."
+              aria-label="The last saved values will be updated"
+              isInline
+            />
+          </>
+        )}
+      </FormGroup>
+    </Form>
+  );
+};
+
+export const StorageClassEncryptionKMSID: React.FC<ProvisionerProps> = ({ onParamChange }) => {
+  const csiCMWatchResource: WatchK8sResource = {
+    kind: ConfigMapModel.kind,
+    namespaced: true,
+    isList: false,
+    namespace: CEPH_STORAGE_NAMESPACE,
+    name: KMSConfigMapCSIName,
+  };
+
+  const [csiConfigMap, csiConfigMapLoaded] = useK8sWatchResource<ConfigMapKind>(csiCMWatchResource);
+
+  React.useEffect(() => {
+    if (csiConfigMapLoaded && csiConfigMap) {
+      const serviceNames: string[] = Object.keys(csiConfigMap?.data);
+      const targetServiceName: string = serviceNames[serviceNames.length - 1];
+      const delimiterIndex: number = targetServiceName.indexOf('-');
+      onParamChange(targetServiceName.slice(delimiterIndex + 1));
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [csiConfigMap]);
+
+  return <></>;
```

Comment:
I cannot understand the use of this component, if this returns nothing.

Reply:
This is because we need to send a value without rendering anything....since plugin accepts a component only, its like this.